digdag-standards/src/main/java/io/digdag/standards/operator/td/TdWaitOperatorFactory.java

```diff
@@ -81,42 +78,18 @@ private TdWaitOperator(Path workspacePath, TaskRequest request)
             this.priority = params.get("priority", int.class, 0);  // TODO this should accept string (VERY_LOW, LOW, NORMAL, HIGH VERY_HIGH)
             this.jobRetry = params.get("job_retry", int.class, 0);
             this.state = request.getLastStateParams().deepCopy();
-            this.existingJobId = state.getOptional(JOB_ID, String.class);
         }
 
         @Override
         public TaskResult runTask()
         {
             try (TDOperator op = TDOperator.fromConfig(params)) {
 
-                // Start the query job
-                if (!existingJobId.isPresent()) {
-                    String jobId = startJob(op);
-                    state.set(JOB_ID, jobId);
-                    throw TaskExecutionException.ofNextPolling(JOB_STATUS_API_POLL_INTERVAL, ConfigElement.copyOf(state));
-                }
-
-                // Poll for query job status
-                assert existingJobId.isPresent();
-
-                TDJobOperator job = op.newJobOperator(existingJobId.get());
-                TDJob jobInfo = job.getJobInfo();
-                TDJob.Status status = jobInfo.getStatus();
-                logger.debug("poll job status: {}: {}", existingJobId.get(), jobInfo);
-
-                // Wait some more if the job is not yet finished
-                if (!status.isFinished()) {
-                    throw TaskExecutionException.ofNextPolling(JOB_STATUS_API_POLL_INTERVAL, ConfigElement.copyOf(state));
-                }
-
-                // Fail the task if the job failed
-                if (status != SUCCESS) {
-                    String message = jobInfo.getCmdOut() + "\n" + jobInfo.getStdErr();
-                    throw new TaskExecutionException(message, ConfigElement.empty());
-                }
+                TDJobOperator job = op.runJob(state, POLL_JOB, this::startJob);
+                state.remove(POLL_JOB);
```

Comment:
Why do you remove this state? Isn't it safer to left it so that `runJob()` can avoid re-run even when an unexpected retry occurs?